stdlib/public/core/Map.swift

```diff
@@ -26,11 +26,10 @@ public struct LazyMapGenerator<
   ///   since the copy was made, and no preceding call to `self.next()`
   ///   has returned `nil`.
   public mutating func next() -> Element? {
-    let x = _base.next()
-    if x != nil {
-      return _transform(x!)
+    guard let x = _base.next() else {
+          return nil
     }
-    return nil
```

Comment:
IMHO a cleaner form for this simple expression (one statement for each branch) would be:

```swift
if let x = _base.next() {
    return _transform(x)
} else {
    return nil
}
```

This clarifies which branch does which.